integration-tests/bats/helper/query-server-common.bash

```diff
@@ -21,12 +21,12 @@ import os
 import sys
 
 args = sys.argv[sys.argv.index('--') + 1:]
-working_dir, database, port_str, timeout_ms = args
+working_dir, database, port_str, timeout_ms, user = args
 os.chdir(working_dir)
 
 from pytest import wait_for_connection
-wait_for_connection(port=int(port_str), timeout_ms=int(timeout_ms), database=database, user='dolt')
-" -- "$PYTEST_DIR" "$DEFAULT_DB" "$1" "$2"
+wait_for_connection(port=int(port_str), timeout_ms=int(timeout_ms), database=database, user=user)
+" -- "$PYTEST_DIR" "$DEFAULT_DB" "$1" "$2" "${SQL_USER:-dolt}"
```

Comment:
Not clear we need the additional user argument. Is this because there are many tests which don't use the `dolt` user?

Reply:
It's because currently commands that don't supply a user but end up using the remote queryist assume a default user of "root", whereas the `start_sql_server` helper function previously assumed a user of "dolt". `local-remote.bash` will set this env variable to `root` when running against a server process.